examples/evaluate.js

```diff
@@ -10,11 +10,13 @@ export default function() {
 
   page.goto('https://test.k6.io/', { waitUntil: 'load' });
   const dimensions = page.evaluate(() => {
-    return {
+    const obj = {
       width: document.documentElement.clientWidth,
       height: document.documentElement.clientHeight,
       deviceScaleFactor: window.devicePixelRatio
     };
+    console.log(obj); // tests #120
```

Comment:
A nitpick perhaps, but I think it'd be better if we separated examples and end-to-end tests, or perhaps moved the `examples/` folder to `tests/integration` or some such (and keep examples to the README and docs). I get the DRYness of not duplicating the JS code, but mixing code meant to just exercise some piece of code for test purposes with code that is meant to examplify usage of the API is not ideal IMO.

Reply:
Yeah, I kind of agree with you, and they were originally separate in #45, but after discussion with Inanc and some minor changes with how we specify `headless`, we settled on reusing them. Right now the differences between both would be minimal as it would just be a 1:1 copy. This logging change is a very minor test (and not very thorough; it basically checks that it doesn't fail), but for proper E2E tests we'd ideally want some way of making assertions about the output, maybe write them in Go as we do for the Cloud, etc.

So can we agree to leave this as is and when we need to add more complex E2E tests that are no longer just examples, to do the split then? I can create a `low prio` issue about it if you want.

Reply:
For example, I converted an example test in JS to a Go version yesterday. Maybe we can write those tests like this:

https://github.com/grafana/xk6-browser/blob/28e81a61d318adf71eb68edcedf2af91f83181dc/tests/context_closed_test.go#L30-L52

Reply:
@inancgumus Those are really integration tests which is mostly what we currently have, but proper E2E tests would involve executing the `xk6-browser` binary as a user would, which is what the `e2e.yml` workflow does. So while there is a distinction between examples and E2E tests, our current E2E tests are not very complex for them to be split up IMO.

That said, I should probably add an integration test in that form that calls `Page.Evaluate()`, as well as a few other input variations to the current ones. I'll do that tomorrow, but you can review the code before that.

Reply:
Hmm, I got your point. So why are we going to type them in Go instead of running them as javascript tests? We can copy them to another directory as E2E tests, unlike what we agreed on before. What's the harm in doing so?

Reply:
BTW, I abandoned the `Page.Evaluate()` test I mentioned above, and only added an additional `ConsoleLogFormatter` one in e2015ec. We can't really assert on the `console.log()` output since we create a new `logrus.Logger` instance and can't hook into it in the test. So this will have to remain an E2E test.